bindings/azure/cosmosgraphdb/cosmosgraphdb.go

```diff
@@ -0,0 +1,120 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation and Dapr Contributors.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package cosmosgraphdb
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/kit/logger"
+	gremcos "github.com/supplyon/gremcos"
+)
+
+const (
+	queryOperation bindings.OperationKind = "query"
+
+	// keys from request's metadata
+	commandGremlinKey = "gremlin"
+
+	// keys from response's metadata
+	respGremlinKey   = "gremlin"
+	respOpKey        = "operation"
+	respStartTimeKey = "start-time"
+	respEndTimeKey   = "end-time"
+	respDurationKey  = "duration"
+)
+
+// CosmosGraphDB allows performing state operations on collections
+type CosmosGraphDB struct {
+	metadata *cosmosGraphDBCredentials
+	client   *gremcos.Cosmos
+	logger   logger.Logger
+}
+
+type cosmosGraphDBCredentials struct {
+	URL       string `json:"url"`
+	MasterKey string `json:"masterKey"`
+	Username  string `json:"username"`
+}
+
+// NewCosmosGraphDB returns a new CosmosGraphDB instance
+func NewCosmosGraphDB(logger logger.Logger) *CosmosGraphDB {
+	return &CosmosGraphDB{logger: logger}
+}
+
+// Init performs CosmosDB connection parsing and connecting
+func (c *CosmosGraphDB) Init(metadata bindings.Metadata) error {
+	m, err := c.parseMetadata(metadata)
+	if err != nil {
+		return errors.New("failed to create the Cosmos Graph DB connector - metadata error")
```

Comment:
Should we be eating the errors here? Will this make debugging more difficult?

Reply:
@halspang  I am not sure, I just trying to be consistent with the other bindings code by following the same patterns. I saw this on the cosmosDB binding code. please let me knew if I need to remove.

Reply:
This is fine